resources/splunk_install.rb

```diff
@@ -0,0 +1,214 @@
+# Encoding: utf-8
+#
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_install
+#
+# Resource for managing the installation of Splunk
+class SplunkInstall < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :version, String, required: true
+  property :build, String, required: true
+  property :base_url, String, default: 'https://download.splunk.com/products'
+
+  default_action :install
+
+  def after_created
+    node.run_state['splunk_ingredient'] ||= { 'installations' => {} }
+
+    case name.downcase
+    when 'splunk' then package :splunk
+    when 'universal_forwarder' then package :universal_forwarder
+    else raise 'Package must be specified (:splunk or :universal_forwarder)'
+    end unless property_is_set? :package
+  end
+
+  ### Inherited Methods
+
+  # Must be overridden by a platform provider
+  def kernel_string
+    raise 'No implementation for current platform'
+  end
+
+  def package_name
+    @package_name ||= package_names[package][node['os'].to_sym]
+    raise "Unsupported Combination: #{package} + #{node['os']}" unless @package_name
+    @package_name
+  end
+
+  def install_dir
+    @install_dir ||= default_install_dirs[package][node['os'].to_sym]
+    raise "Unsupported Combination: #{package} + #{node['os']}" unless @install_dir
+    @install_dir
+  end
+
+  def package_url
+    @package_url ||=
+      case package
+      when :splunk then "#{base_url}/splunk/releases/#{version}/#{node['os']}/splunk"
+      when :universal_forwarder then "#{base_url}/universalforwarder/releases/#{version}/#{node['os']}/splunkforwarder"
+      end + "-#{version}-#{build}-#{kernel_string}"
+  end
+
+  def package_path
+    @package_path ||= Pathname.new(Chef::Config['file_cache_path']) + filename_from_url(package_url)
+  end
+
+  ### Inherited Actions
+
+  load_current_value do |desired|
+    package desired.package
+    version_file = version_pathname(install_dir)
+    current_value_does_not_exist! unless version_file.exist?
+
+    version_data = version_file.read
+    installed_version, installed_build = version_data.match(/VERSION=(\d(?:\.\d)+).+BUILD=([\w]+)/m).captures
+    version installed_version
+    build installed_build
+  end
+
+  action :install do
+    install_state = node.run_state['splunk_ingredient']
+    raise "Install at #{install_dir} already exists!" if install_state['installations'][install_dir]
```

Comment:
Yes, executing the install action multiple times pointing to the same installation path (currently, basically the same package) will throw this error. You can run any other actions (like uninstall) without encountering this error, but attempting to install it twice will be an error.

It should be possible to run the install, uninstall, then install in the same chef run without encountering this error, but that's weird and isn't tested.